app/gui/src/dashboard/layouts/AssetsTable.tsx

```diff
@@ -250,6 +252,7 @@ function AssetsTable(props: AssetsTableProps) {
       refetchInterval: listDirectoryRefetchInterval,
     }),
     retry: () => {
+      if (currentDirectoryId !== currentDirectoryIdRef.current) return true
```

Comment:
I don't understand why this is needed. Does the result of the query depend on `currentDirectoryId` somehow, in a way that is not covered by the `queryKey`?

Reply:
Ok. I don't see why this side effect is performed from the `retry` logic--the `onFailure` function will be run if and only if `retry` returns false, so couldn't the `setDriveLocation` call be there?

Reply:
`on*` doesn't seem to exist on `useSuspenseQuery` unfortunately?